### ISSUE TITLE
View Bobbing gets disabled but not re-enabled

### ISSUE THREAD

**Issue by FireEgl** [Thutmose/Thut#6]
#### Issue Description:

Stepping onto an elevator disables view bobbing, and stepping off doesn't re-enable it..
#### What happens:

When you step onto an elevator, view bobbing (in your minecraft options) gets turned off,
when you step off the elevator, view bobbing doesn't get turned back on.
#### What you expected to happen:

I expect view bobbing to not be touched, or at least turned back on when I get off the elevator.
#### Steps to reproduce:
1. Make sure view bobbing is turned on.
2. step onto an elevator.
3. check your view bobbing setting to see that it got turned off.
   It doesn't matter if you leave the elevator, view bobbing remains off.

---
#### Affected Versions (Do _not_ use "latest"):
- Thut Core: thutcore-1.9.4-3.1.10.jar
- Thut's Elevators (If Applicable): thuttech-3.0.5.jar
- Forge: 12.18.0.2008

**Comment by Thutmose** [comment]
this is currently intentional, View bobbing causes issues with jitteriness in the elevator.

Detecting when a player has gotten off the elevator is not as easy as detecting when a player is on it, so for performance reasons it just turns off bobbing.  You can disable this in configs, but it will result in slight jitteriness when standing on the elevator.

**Comment by MechWarrior99** [comment]
I was going to post an issue about this, but then I saw this. 
I'm no coder, but I was wondering, what if you had it so the elevator knew how big it was, say 3x3. And then did a radius check around it for players in a radius of 5x5 or 7x7, and any players that are not in the 3x3 but are in the 5x5 or 7x7 (idk which would be better) would have bobbing turned on?  

like I said, I'm no coder, but I thought I would through this idea out.

**Comment by Thutmose** [comment]
The difficulty would be knowing if they are supposed to have view bobbing turned on.

I never play with it on, as view bobbing is very un-immersive for me, my view irl does not bob (brain accounts for that when walking...), so why should it in game?  So having it randomly turn it on for me will be very annoying.

**Comment by MechWarrior99** [comment]
Yeah, I understand that, I know some people love view bobbing on and some hate it.  

There is probably no way to do a check to see if view bobbing has been turned off in the past (as an example) 5 minutes, is there?  Or maybe see if when the player joined the world if view bobbing was on or off?

**Comment by brisingraerowing** [comment]
Is it possible for the mod to see if the player has it turned off by themselves (e.g. it is off when they first get onto an elevator)? Or would a whitelist of players who want it turned back on work?

**Comment by Thutmose** [comment]
see if this version is any help? https://minecraft.curseforge.com/projects/thutcore/files/2343065

**Comment by hiimcody1** [comment]
I am using the thuttech-4.0.9.jar build that appears to include the "jitterfix" config option. I have it set to false on both the client and server yet it still forces off view bobbing. Is the build you referenced above the only one with this config flag active?

**Comment by Thutmose** [comment]
the fix was in thutcore, rather than thuttech, I don't think the jitterfix option does anything anymore.

**Comment by rakarukri** [comment]
Hi Thutmose!

I already change the fixJitter state to false in thuttech.cfg. But, when I step into the elevator the bobbing still changed to off even I walked out from the elevator. I'm using thuttech-5.0.0 and thutcore-4.3.1. Can you fix this please?

Thank you.

**Comment by ghost** [comment]
I was going to create an issue for this and saw it was already done. I have a whole modpack so I wasn't sure elevators was the cause of the issue, even though I suspected it was. Glad I'm not crazy lol, the first time it happened I was so confused. Like "wtf? why is my view bobbing off?" hahaha.
It would be awesome if this was fixed, even though I understand it may be difficult. I like playing with view bobbing on.

**Comment by ghost** [comment]
Question, though: would it be easier to set it so that view bobbing gets turned on/off depending on when the elevator is moving? Most likely if the elevator is not moving it means the player has gotten off. So can you just have it turn view bobbing off when the elevator button gets pressed to go up/down, and have it turn back on when the elevator is done moving? That way you don't need to detect where the player is, just whether the elevator is moving (and whether the player has view bobbing on, I guess)

**Comment by Thutmose** [comment]
the view bobbing issue affects any player who is even standing on the elevator, ie if you walk across it while it is stationary, it will cause jitteriness in your view.  I had a partial fix in, but I suspect it may have been disabled when I re-wrote the collision code to fix some other issues, I can look into trying to revive the fix when I get time, it will probably be done as a client side config option like it was before
